prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/kubeapi/KubeapiWorkFlowTaskTest.java

```diff
@@ -0,0 +1,120 @@
+package com.redhat.parodos.tasks.kubeapi;
+
+import com.redhat.parodos.workflow.context.WorkContextDelegate;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.util.generic.dynamic.DynamicKubernetesObject;
+import io.kubernetes.client.util.generic.dynamic.Dynamics;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+public class KubeapiWorkFlowTaskTest {
+
+	private final static ApiClient apiClient = Mockito.mock(ApiClient.class);
+
+	private final static KubernetesApi api = Mockito.mock(KubernetesApi.class);
+
+	private final static KubeapiWorkFlowTask task = new KubeapiWorkFlowTask(apiClient, api, "Test");
+
+	@Test
+	public void missingArgs() {
+		WorkContext ctx = createWorkContext(new HashMap<>());
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(MissingParameterException.class, result.getError().getClass());
+	}
+
+	@Test
+	public void invalidOperationType() {
+		HashMap<String, String> map = new HashMap<>();
+		map.put("operation", "test");
+		WorkContext ctx = createWorkContext(map);
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(IllegalArgumentException.class, result.getError().getClass());
+	}
+
+	@Test
+	public void get() throws ApiException {
+		String objJson = "{\"kind\":\"ConfigMap\",\"apiVersion\":\"v1\",\"metadata\":{\"name\":\"my-cm\",\"namespace\":"
+				+ "\"test\",\"uid\":\"5f28eebc-05b2-41b0-9da1-e7399adc1262\",\"resourceVersion\":\"163811\","
+				+ "\"creationTimestamp\":\"2023-04-16T18:47:25Z\"},\"data\":{\"a\":\"vala\",\"b\":\"valb\"}}";
+		DynamicKubernetesObject obj = Dynamics.newFromJson(objJson);
+		WorkContext ctx = workContextGET();
+		doReturn(obj).when(api).get(any(), any(), any(), any(), any(), any());
```

Comment:
Please use the parameters referenced to the configmap that are expected for the call instead of `any()`.
In this case you can use the `apiGroup, apiVersion, kindPluralName,namespace and name` from the manifest itself.

Reply:
we do it in other tests as well. Actually, Eloy suggested I use it in my previous assignment. It makes the test coe much easier to manage.

Reply:
Here you are asserting that the parameters that you pass to this method will match certain values. By specifying those values in the mock, you ensure that they are what you expect and not something else, such as empty, null or even passed in the wrong order. 
By asserting for `any()` you miss the opportunity to validate this scenario and makes the test less trustworthy.

Reply:
I am not trying to assert/validate these values. there's no point using anything but any()

Reply:
I see. So it's on purpose. Please add the validations as I mentioned, otherwise this test is not complete.